File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/viewmodel/WanHomeViewModel.java
package com.yyxnb.module_wanandroid.viewmodel;

import android.arch.lifecycle.MutableLiveData;

import com.yyxnb.common_base.base.CommonViewModel;
import com.yyxnb.common_base.bean.WanData;
import com.yyxnb.common_base.config.Http;
import com.yyxnb.module_wanandroid.bean.WanAriticleBean;
import com.yyxnb.module_wanandroid.bean.WanStatus;
import com.yyxnb.module_wanandroid.config.WanService;

import java.util.List;

public class WanHomeViewModel extends CommonViewModel {

    private final WanService mApi = Http.getInstance().create(WanService.class);

    public MutableLiveData<List<WanAriticleBean>> bannerData = new MutableLiveData<>();
    public MutableLiveData<List<WanAriticleBean>> topArticleData = new MutableLiveData<>();
    public MutableLiveData<WanStatus<WanAriticleBean>> homeListData = new MutableLiveData<>();

    public void getAritrilList(int page){

        launchOnlyResult(mApi.getAritrilList(page), new OnHandleException<WanData<WanStatus<WanAriticleBean>>>() {
            @Override
            public void success(WanData<WanStatus<WanAriticleBean>> data) {
                homeListData.postValue(data.getResult());
            }

            @Override
            public void error(String msg) {
                loge(msg);
            }
        });
    }

    public void getTopAritrilList(){

        launchOnlyResult(mApi.getTopAritrilList(), new OnHandleException<WanData<List<WanAriticleBean>>>() {
            @Override
            public void success(WanData<List<WanAriticleBean>> data) {
                topArticleData.postValue(data.getResult());
            }

            @Override
            public void error(String msg) {
                loge(msg);
            }
        });
    }

    public void getBanner(){

        launchOnlyResult(mApi.getBanner(), new OnHandleException<WanData<List<WanAriticleBean>>>() {
            @Override
            public void success(WanData<List<WanAriticleBean>> data) {
                bannerData.postValue(data.getResult());
            }

            @Override
            public void error(String msg) {
                loge(msg);
            }
        });
    }

}

File: module_wanandroid/src/main/java/com/yyxnb/module_wanandroid/viewmodel/WanHomeViewModel.kt
package com.yyxnb.module_wanandroid.viewmodel

import androidx.lifecycle.MutableLiveData
import com.yyxnb.common.utils.log.LogUtils
import com.yyxnb.common_base.config.Http
import com.yyxnb.module_wanandroid.bean.WanAriticleBean
import com.yyxnb.module_wanandroid.bean.WanStatus
import com.yyxnb.module_wanandroid.config.WanService
import com.yyxnb.network.BaseViewModel

class WanHomeViewModel : BaseViewModel() {

    private val mApi = Http.create(WanService::class.java)

    @JvmField
    val bannerData = MutableLiveData<List<WanAriticleBean>>()

    @JvmField
    val topArticleData = MutableLiveData<List<WanAriticleBean>>()

    @JvmField
    val homeListData = MutableLiveData<WanStatus<WanAriticleBean>>()

    fun getAritrilList(page: Int) {
        launchOnlyResult(
                block = { mApi.getAritrilList(page) },
                reTry = { getAritrilList(page) },
                success = { homeListData.postValue(it) },
                error = { LogUtils.e(it.message) }
        )
    }

    fun getTopAritrilList() {
        launchOnlyResult(
                block = { mApi.topAritrilList() },
                reTry = { getTopAritrilList() },
                success = { topArticleData.postValue(it) },
                error = { LogUtils.e(it.message) }
        )
    }


    fun getBanner() {
        launchOnlyResult(
                block = { mApi.banner() },
                reTry = { getBanner() },
                success = { bannerData.postValue(it) },
                error = { LogUtils.e(it.message) }
        )
    }

}